src/FastExpressionCompiler/FastExpressionCompiler.cs

```diff
@@ -2693,15 +2695,20 @@ private static void EmitValueTypeDereference(ILGenerator il, Type type)
                     return false;
 
                 if ((parent & ParentFlags.IgnoreResult) != 0)
-                    il.Emit(OpCodes.Pop);
-                else
+                    return true;

```

Comment:
Ah, this is actually another bugfix for ignored results - I discovered this while just poking around to fix #310. Since the parent flags are propagated to TryEmit, there is nothing on the stack, so we also need to pop.